Revit_Core_Engine/Create/AddInsulation.cs

```diff
@@ -0,0 +1,108 @@
+        /***************************************************/
+
+        [Description("Add or update existing insulation of the duct or pipe with given inuslation type and thickness.")]
+        [Input("host", "Host element to add the insulation on.")]
+        [Input("insulationType", "Type of insulation to be added on the host object.")]
+        [Input("insulationThickness", "Thickness of insulation to be added on the host object in millimeters.")]
+        [Output("insulation", "Insulation that has been added on the host.")]
+        public static InsulationLiningBase AddInsulation(this Element host, ElementType insulationType, double insulationThickness)
+        {
+            Document doc = host.Document;
+            //Check inputs
+            if (insulationType == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Insulation could not be created. Insulation type not found.");
+                return null;
+            }
+
+            if (host == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Insulation could not be created. Host element not found.");
+                return null;
+            }
+
+            InsulationLiningBase ins;
+            ICollection<ElementId> insIds = InsulationLiningBase.GetInsulationIds(doc, host.Id);
+            insulationThickness = insulationThickness / 304.8;
+            BuiltInCategory hostCategory = (BuiltInCategory)host.Category.Id.IntegerValue;
+            if (insIds.Count == 0)
+            {
+                if (host is Pipe || hostCategory is BuiltInCategory.OST_PipeFitting || hostCategory is BuiltInCategory.OST_PipeAccessory || hostCategory is BuiltInCategory.OST_FlexPipeCurves)
+                {
+                    ins = PipeInsulation.Create(doc, host.Id, insulationType.Id, insulationThickness);
+                }
+                else if (host is Duct || hostCategory is BuiltInCategory.OST_DuctFitting || hostCategory is BuiltInCategory.OST_DuctAccessory || hostCategory is BuiltInCategory.OST_FlexDuctCurves)
+                {
+                    ins = DuctInsulation.Create(doc, host.Id, insulationType.Id, insulationThickness);
+                }
+                else
+                {
+                    BH.Engine.Reflection.Compute.RecordError("Insulation could not be created. Host element type does not support insulation adding functionality.");
+                    return null;
+                }
+            }
+            else if (insIds.Count == 1)
+            {
+                ElementId insId = insIds.ToList()[0];
+                ins = doc.GetElement(insId) as InsulationLiningBase;
+                if (ins.GetTypeId() != insulationType.Id)
+                {
+                    ins.ChangeTypeId(insulationType.Id);
+                }
+                if (ins.Thickness != insulationThickness)
+                {
+                    ins.Thickness = insulationThickness;
+                }
```

Comment:
This could be a oneliner:
```suggestion
                ins.Thickness = insulationThickness;
```